.github/workflows/python_tests.yml

```diff
@@ -47,19 +42,39 @@ jobs:
 
       # TODO: remove --no-skip-cached-tests if/when we cache coverage
       # <https://github.com/kkrt-labs/keth/issues/694>
-      - name: Run tests without caching
+      - name: Run unit tests without caching
         run: |
           uv run --reinstall pytest -n logical --durations=0 -v -s --log-cli-level=DEBUG --no-skip-cached-tests --ignore-glob=cairo/tests/ef_tests/
 
       - uses: actions/cache/save@v4
         with:
           path: .pytest_cache
           key: ${{ runner.os }}-pytest-cache-${{ github.run_id }}
-
       - name: Upload coverage
         uses: codecov/codecov-action@v5
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           verbose: true
           files: ./coverage/**/*.json
-          flags: ci
+          flags: ci-unit
+
+  ef-tests-sampled:
+    name: EF-Tests Sampled
+    runs-on: ubuntu-latest-64-cores
+    env:
+      HYPOTHESIS_PROFILE: ci
+      CAIRO_PATH: cairo:tests:python/cairo-ec/src:python/cairo-addons/src:python/cairo-core/src
+    steps:
+      - uses: actions/checkout@v4
+      - uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-dependency-glob: uv.lock
+      - uses: actions/setup-python@v5
+        with:
+          python-version-file: .python-version
+      - name: Run sampled EF tests
```

Comment:
should we add codecov as well?

Reply:
If it takes car of the max 2000 and radomization of tests it would be great but seems tricky.